cmd/opentelemetry-collector/app/defaults/default_config_test.go

```diff
@@ -141,22 +148,45 @@ func TestDefaultCollectorConfig(t *testing.T) {
 }
 
 func TestDefaultAgentConfig(t *testing.T) {
-	v, _ := jConfig.Viperize(grpc.AddFlags)
-	factories := Components(v)
-	cfg := AgentConfig(factories)
-	assert.Equal(t, configmodels.Service{
-		Extensions: []string{"health_check"},
-		Pipelines: configmodels.Pipelines{
-			"traces": &configmodels.Pipeline{
-				InputType: configmodels.TracesDataType,
-				Receivers: []string{"jaeger"},
-				Exporters: []string{"jaeger"},
+	tests := []struct {
+		config  map[string]interface{}
+		service configmodels.Service
+	}{
+		{
```

Comment:
Sorry should have mentioned in preview review - there needs to be a test without the flag being specified to confirm `resource` processor is not added.

Reply:
that was my intention when I refactored the test to use arrays, but I apparently forgot to add the second use case.